src/components/Body/Trials/TrialRow/TrialRow.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useRef, useEffect } from 'react';
+import Buttons from '../../../Common/Buttons/Buttons';
+import Circles from './Circles/Circles';
+import NumberInput from './NumberInput/NumberInput';
+import classes from './TrialRow.module.css';
+
+
+const TrialRow = (props) => {
+
+    //STATES & HOOKS------------------------------------------------------------------
+    const [correctNumbersWithIndex, setCorrectNumbersWithIndex] = useState(0);
+    const [correctNumbersOnly, setCorrectNumbersOnly] = useState(0);
+    const [rowEnabled, setRowEnabled] = useState(false);
+    const enteredNumberRef = useRef([]); //to get the entered number from the number input 
+    const numberOfDigits = 4;
+    const numberOfTrials = 8;
```

Comment:
same here, please move them to the service file

Reply:
done